src/vm.rs

```diff
@@ -54,6 +54,26 @@ pub trait SyscallObject<E: UserDefinedError> {
     ) -> Result<u64, EbpfError<E>>;
 }
 
+/// A virtual method table for SyscallObject
```

Comment:
These look jit specific and not part of the VM API, any reason not to put them into `jit.rs`?